From dd33cc6b24aaca3b2078a7b05f4d8a31aa7084af Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Fri, 18 May 2012 18:11:23 +0900
Subject: [PATCH 14/80] rcarvin: reset the occupied buffers counter in remove_device()

The occupied buffers counter 'mb_cnt' could be incremented when
a buffer would be set up to be filled in the irq handler, and
decremented when one of the buffers would be released in
videobuf_release().

Unfortunatery videobuf_release() invocation could be skipped
if a user application is terminated.
This change must be required for such above situation.

Signed-off-by: Katsuya Matsubara <matsu@igel.co.jp>
---
 drivers/media/video/rcarvin.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index e72b3aa..651015c 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -765,6 +765,9 @@ static void rcar_vin_remove_device(struct soc_camera_device *icd)
 		}
 	}
 
+	/* reset mb_cnt since all of the buffers were released. */
+	pcdev->mb_cnt = 0;
+
 	spin_unlock_irqrestore(&pcdev->lock, flags);
 
 #ifdef CONFIG_PM
-- 
1.7.0.4

